lib/bigcommerce/lightstep/mysql/adapter.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+# Copyright (c) 2020-present, BigCommerce Pty. Ltd. All rights reserved
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+# OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+module Bigcommerce
+  module Lightstep
+    module Mysql
+      ##
+      # Patches mysql and ActiveRecord to allow for mysql span tracing
+      #
+      module Adapter
+        extend ::ActiveSupport::Concern
+
+        ##
+        # Patch ActiveRecord to enable mysql span traces
+        #
+        def self.patch
+          return unless enabled?
+
+          # rubocop:disable Lint/SendWithMixinArgument
+          ::ActiveRecord::ConnectionAdapters::Mysql2Adapter.send(:include, ::Bigcommerce::Lightstep::Mysql::Adapter)
```

Comment:
What are your thoughts on making this namespaced as an ActiveRecord adapter instead of a generic MySQL one? I'd like to explore [rom-rb](https://rom-rb.org/) in the future and having a structure to add a new, non-ActiveRecord MySQL adapter would be nice.

Reply:
I think that's fair. I'll make the change.